snapcraft/projects.py

```diff
@@ -415,6 +435,23 @@ def _validate_default_provider(cls, default_provider):
         return default_provider
 
 
+class Component(ProjectModel):
+    """Snapcraft component definition."""
+
+    summary: ProjectSummary
+    description: str
+    type: Literal["test"]
+    version: Optional[ProjectVersion]
+
+    @pydantic.validator("version")
+    @classmethod
+    def _validate_version(cls, version):
+        if version:
```

Comment:
Do we want to allow an empty string in `version`? If not,

```suggestion
        if version is not None:
```

Reply:
Good catch, I added a check and an extra test.